Source/Core/Core/IOS/WFS/WFSSRV.cpp

```diff
@@ -246,6 +365,19 @@ IPCCommandResult WFSSRV::IOCtl(const IOCtlRequest& request)
   return GetDefaultReply(return_error_code);
 }
 
+s32 WFSSRV::Rename(const std::string& source, const std::string& dest) const
+{
+  const bool opened = std::any_of(m_fds.begin(), m_fds.end(), [&](const auto& fd) {
+    return fd.in_use && fd.path == source;
+  });
+
+  if (opened)
+    return WFS_FILE_IS_OPENED;
+
+  File::Rename(WFS::NativePath(source), WFS::NativePath(dest));
```

Comment:
Errors are pretty unlikely to happen (read/write failures aren't checked either), so I didn't bother checking for failure here. I guess we could handle the most common failure case (ENOENT) though?